### PR TITLE
Add `manage_data_stream_lifecycle` to built-in privs test

### PR DESCRIPTION
**Pull request by n1v0lg**

### CODE REVIEW

File: x-pack/docs/build.gradle [hunk 1]
```diff
@@ -29,18 +27,6 @@ restResources {
...
 
-// TODO: Remove the following when the following features are released. These tests include new privileges only available under feature flags
-//  which require snapshot builds:
-// * Data Stream Lifecycle. manage_data_stream_lifecycle privilege is only available with dlm_feature_flag_enabled set
-// We disable these tests for snapshot builds to maintain release build coverage.
-tasks.named("yamlRestTest").configure {
-  if (BuildParams.isSnapshotBuild()) {
-    systemProperty 'tests.rest.blacklist', '*/get-builtin-privileges/*'
-  } else {
-    systemProperty 'tests.rest.blacklist', ['*/create-cross-cluster-api-key/*', '*/update-cross-cluster-api-key/*'].join(',')
```
Comment (n1v0lg) [inline x-pack/docs/build.gradle:40]: @ywangd data stream lifecycle is not behind a feature flag anymore so I'm removing that part. Sanity-check: it should also be fine to remove the RCS-related bits here, correct?

**Review by andreidan — APPROVED** [review]
❤️ I was just about to do this :) 

Thanks Nikolaj 🚀

File: x-pack/docs/build.gradle (hunk 1 above)
Comment (ywangd) [reply x-pack/docs/build.gradle:40]: Thanks for catching this. Yes they should have been removed. Thanks!

Is it possible to remove them for 8.10 as well? Is this PR backportable?

Comment (n1v0lg) [reply x-pack/docs/build.gradle:40]: cool thanks for confirming. 

I don't think we can backport as is: https://github.com/elastic/elasticsearch/pull/98644 only removed feature flag for lifecycles in 8.11. 

I'll just open a separate PR fixing RCS 2.0 and backport -- and then do this PR without RCS changes.

Comment (n1v0lg) [reply x-pack/docs/build.gradle:40]: @ywangd PR here: https://github.com/elastic/elasticsearch/pull/99038

**Comment by n1v0lg** [comment]
@elasticmachine update branch plz

**Comment by n1v0lg** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Pull request merged**